Backends/CLX/text-selection.lisp

```diff
@@ -97,20 +97,23 @@
   (setf (selection-timestamp port) nil))
 
 (defmethod request-selection ((port clx-text-selection-port-mixin) requestor time)
-  (xlib:convert-selection :primary :STRING requestor :bounce time))
+  (xlib:convert-selection :primary :UTF8_STRING requestor :bounce time))
 
 (defmethod get-selection-from-event ((port clx-text-selection-port-mixin) (event clx-selection-notify-event))
-  ; (describe event *trace-output*)  
   (if (null (selection-event-property event))
       (progn
         (format *trace-output* "~&;; Oops, selection-notify property is null. Trying the cut buffer instead..~%")
         (xlib:cut-buffer (clx-port-display port)))                
-      (map 'string #'code-char
-           (xlib:get-property (sheet-xmirror (event-sheet event))
-                              (selection-event-property event)
-                              ;; :type :text
-                              :delete-p t
-                              :result-type 'vector))))
+      (let* ((v (xlib:get-property (sheet-xmirror (event-sheet event))
+                                   (selection-event-property event)
+                                   ;; :type :text
+                                   :delete-p t
+                                   :result-type '(vector (unsigned-byte 8))))
+             (type (case (clim-clx::selection-event-target event)
```

Comment:
this whole let block can be written as:
```lisp
(let ((v (xlib:get-property (sheet-xmirror (event-sheet event))
                                (selection-event-property event)
                                  ;; :type :text
                                   :delete-p t
                                   :result-type '(vector (unsigned-byte 8)))
    (ccase (clim-clx::selection-event-target event)
      (:string (babel:octets-to-string v :encoding :iso-88519-1))
      (:utf8_string (babel:octets-to-string v :encoding :utf-8))))
```

Reply:
I wanted to avoid specifying `octets-to-string` multiple times, but I can see how your solution is actually more concise.

Fixed in: d95035a0ca776078f1fde6dab941102d4d4e4c19

Note that I changed your use of `CCASE` to `CASE`, since there is no setf function for that form.